presto-hive/src/main/java/com/facebook/presto/hive/HivePartitionManager.java

```diff
@@ -148,19 +142,11 @@ public HivePartitionResult getPartitions(SemiTransactionalHiveMetastore metastor
         List<String> partitionNames = getFilteredPartitionNames(metastore, tableName, partitionColumns, effectivePredicate);
 
         // do a final pass to filter based on fields that could not be used to filter the partitions
-        int partitionCount = 0;
         ImmutableList.Builder<HivePartition> partitions = ImmutableList.builder();
         for (String partitionName : partitionNames) {
             Optional<Map<ColumnHandle, NullableValue>> values = parseValuesAndFilterPartition(partitionName, partitionColumns, partitionTypes, constraint);
 
             if (values.isPresent()) {
-                if (partitionCount == maxPartitions) {
```

Comment:
I don't think we should change this. The entire reason for the limit is as safety check to prevent GC / OOM problems when users run a query over a table with millions of partitions (those really exist, unfortunately). We only care that the user tries to query a million partitions because it crashes the coordinator.

Reply:
> The entire reason for the limit is as safety check to prevent GC / OOM problems when users run a query over a table with millions of partitions

Such queries are still prevented. I didn't remove the check, just moved it to `HiveSplitManager`